pkg/agent/protocol/http/proxy_test.go

```diff
@@ -331,6 +398,14 @@ func Test_ProxyHandler(t *testing.T) {
 				return
 			}
 
+			// Compare headers only if either expected or returned have items.
+			// We have to check for length explicitly as otherwise a nil map would not be equal to an empty map.
+			if len(tc.headers) > 0 || len(tc.expectedHeaders) > 0 {
+				if diff := cmp.Diff(tc.expectedHeaders, resp.Header); diff != "" {
```

Comment:
It is possible to compare this without `cmp`, (e.g. `reflect.DeepEqual`, or writing headers to a `bytes.Buffer`), but I find that the output of `cmp` (a diff) is very easy to read in test output. Moreover it was already a transitive dependency of the project, so I'll vouch for keeping it (and even start using it more!)

Reply:
Nice addition. I haven't used it before because I haven't realized it was a transitive reference!